habitat_sim/gfx_replay_utils.py

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import magnum as mn
+
+import habitat_sim
+
+
+def add_node_user_transform(sim, node, name):
+    translation = node.absolute_translation
+    rot = mn.Quaternion(mn.Quaterniond(node.rotation))
+    while True:
+        node = node.parent
+        if not node:
+            break
+        try:
+            rot = node.rotation * rot
+        except AttributeError:
+            # scene root has no rotation attribute
+            break
+    sim.gfx_replay_manager.add_user_transform_to_keyframe(name, translation, rot)
+
+
+def make_backend_configuration_for_playback(
+    need_separate_semantic_scene_graph=False,
+):
+    backend_cfg = habitat_sim.SimulatorConfiguration()
+    backend_cfg.scene_id = "NONE"  # see Asset.h EMPTY_SCENE
+    backend_cfg.force_separate_semantic_scene_graph = need_separate_semantic_scene_graph
+
+    return backend_cfg
```

Comment:
Should be in habitat_sim/utils directory, no?

Reply:
Previously, we've discussed that this utils directory should be where we start putting "pure" utilities, I mean, utilities that don't rely on the rest of habitat_sim. We can eventually even modify our CI to enforce this. @aclegg3 is this your understanding of this directory? Perhaps we should rename to "core_utils" ("core" implying few other dependencies)?

In the case if gfx_replay_utils, this is a utility that very much relies on habitat_sim (namely, gfx_replay).

Reply:
Fair enough. Would it be reasonable to find somewhere to put it under habitat-sim/habitat_sim that would help keep things organized within that directory? I'm concerned that habitat-sim/habitat_sim run the risk of becoming a disorganized catch-all.

Reply:
Currently, the role of the utils directory is muddy in this regard. There are modules which require habitat (e.g. profiling_utils.py, viz_utils.py) and those which don't (e.g. common.py, collect_env.py, and compare_profiles.py). Some of these still require Magnum or other libraries. 

I think we agreed that there should be a separation between utilities that require a Habitat-sim install and those which do not. Perhaps this should manifest as separate "habitat-utils" and "utils" directories, or a single directory with per-file separation as exists now. 

Either way, I support John's suggestion that this should be moved somewhere and it feels like "utils" is the right place at this point (it seems like this should maybe live next to profiling and visualization utilities). Perhaps we can put it there and then follow-up with a refactor for a better division later.